Microsoft.Toolkit.Uwp.UI/Shadows/AttachedDropShadow.cs

```diff
@@ -0,0 +1,353 @@
+                    {
+                        mask = ((Shape)context.Element).GetAlphaMask();
+                    }
+                    else if (context.Element is TextBlock)
+                    {
+                        mask = ((TextBlock)context.Element).GetAlphaMask();
+                    }
+                }
+
+                // If we don't have a mask and have specified rounded corners, we'll generate a simple quick mask.
+                // This is the same code from link:AttachedCardShadow.cs:GetShadowMask
+                if (mask == null && SupportsCompositionVisualSurface && CornerRadius > 0)
+                {
+                    // Create rounded rectangle geometry and add it to a shape
+                    var geometry = context.GetResource(RoundedRectangleGeometryResourceKey) ?? context.AddResource(
+                        RoundedRectangleGeometryResourceKey,
+                        context.Compositor.CreateRoundedRectangleGeometry());
+                    geometry.CornerRadius = new Vector2((float)CornerRadius);
+
+                    var shape = context.GetResource(ShapeResourceKey) ?? context.AddResource(ShapeResourceKey, context.Compositor.CreateSpriteShape(geometry));
+                    shape.FillBrush = context.Compositor.CreateColorBrush(Colors.Black);
+
+                    // Create a ShapeVisual so that our geometry can be rendered to a visual
+                    var shapeVisual = context.GetResource(ShapeVisualResourceKey) ??
+                                      context.AddResource(ShapeVisualResourceKey, context.Compositor.CreateShapeVisual());
+                    shapeVisual.Shapes.Add(shape);
+
+                    // Create a CompositionVisualSurface, which renders our ShapeVisual to a texture
+                    var visualSurface = context.GetResource(VisualSurfaceResourceKey) ??
+                                        context.AddResource(VisualSurfaceResourceKey, context.Compositor.CreateVisualSurface());
+                    visualSurface.SourceVisual = shapeVisual;
+
+                    // Create a CompositionSurfaceBrush to render our CompositionVisualSurface to a brush.
+                    // Now we have a rounded rectangle brush that can be used on as the mask for our shadow.
+                    var surfaceBrush = context.GetResource(SurfaceBrushResourceKey) ?? context.AddResource(
+                        SurfaceBrushResourceKey,
+                        context.Compositor.CreateSurfaceBrush(visualSurface));
+
+                    geometry.Size = visualSurface.SourceSize = shapeVisual.Size = context.Element.RenderSize.ToVector2();
+
+                    mask = surfaceBrush;
+                }
+            }
+
+            // Position our shadow in the correct spot to match the corresponding element.
+            context.SpriteVisual.Size = context.Element.RenderSize.ToVector2();
+            context.SpriteVisual.Offset = context.Element.CoordinatesFrom(CastTo).ToVector3();
```

Comment:
@Ryken100 suggested somehow using composition animation to lock the two together? @Sergio0694 thoughts on this too?

Right now I just update the size/offset when the mask is created and in OnSizeChanged below. This seems to work alright for most basic scenarios, but I would imagine would break down with extra translations and if the element casting is animated?

Reply:
Just an idea, but thinking we might be able to use a composition animation to bind the two together like we're doing with `BindSize` for the brushes? I mean not with exactly the same code, but by reusing the same concept:

https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/b60515414660f438db3e82e2f7a1b98b45d5fb07/Microsoft.Toolkit.Uwp.UI.Media/Extensions/Windows.UI.Composition/CompositionObjectExtensions.cs#L25-L34

Reply:
Yeah, that could work for Size, but Offset may not work the same as  we need to do the absolute offset, eh?

Which means we still need the SizeChanged on the parent at least as that effects the offsets...?